monitor/stats.go

```diff
@@ -0,0 +1,49 @@
+package monitor
+
+import (
+	"math"
+
+	"github.com/influxdata/influxdb/monitor/diagnostics"
+	"github.com/influxdata/influxdb/pkg/limiter"
+)
+
+// stats captures statistics
+type stats struct {
+	comp compactThroughputStats
+}
+
+type compactThroughputStats struct {
+	limiter limiter.Rate
+	burst   int
+}
+
+// CompactThroughputUsage calculates the percentage of burst capacity currently consumed by compaction.
+//
+// available = current tokens in the rate limiter bucket (can be negative when in debt)
+// burst = maximum tokens the bucket can hold
+// usage percentage = ((burst - available) / burst) * 100
+func (s *stats) CompactThroughputUsage() float64 {
```

Comment:
This only calculates the burst percentage (maximum tokens that can be requested).  But we also need the overall percentage used of the limit over time ([Limiter.Limit](https://pkg.go.dev/golang.org/x/time/rate#Limiter.Limit)), don't we?

Reply:
I can likely generate two fields

```
compact-throughput-burst-usage

compact-throughput-usage
```

And return the usage of our burst limit and our standard limit?

Reply:
Will see what Andrew's input is

Reply:
Can you use [Limiter.AllowN](https://pkg.go.dev/golang.org/x/time/rate#Limiter.AllowN) to calculate a instantaneous rate check?

Reply:
I believe the correct computation on this is 

`100*(1-rate.Limit(l.Tokens())/l.Limit())`

See [this Go Playground](https://goplay.tools/snippet/tzcypvZAkhX) for an example, and how token debt is handled.